test/supportShim.js

```diff
@@ -0,0 +1,284 @@
+   * @returns {Object}
+   */
+  function examine(method, name) {
+    if (typeof method !== 'function') return;
+
+    // find test hints if provided
+    var obj = hintsModule.full(method.toString(), 'testhint', {function: true}),
+      hints = obj.hints,
+      tree = obj.tree;
+
+    var result = {};
+
+    // extract function arguments
+    var args = getFunctionArguments(tree);
+
+    // create args conform function
+    result.conform = getArgumentsConformFn(method, args, obj.hintsPos, tree);
+
+    // use hints to find index
+    var hint = hints.options;
+    if (hint === 'none') return;
+    if (hint && hint.match(/^\d+$/)) {
+      result.index = hint * 1;
+      return result;
+    }
+
+    // skip if function name does not start with lower case letter
+    if (!name.match(/^[a-z]/)) return;
+
+    // find 'options' argument - if none, then skip
+    var index = args.indexOf('options');
+    if (index === -1) return;
+
+    result.index = index + 1;
+    return result;
+  }
+
+  /*
+   * Shims a method to check for `options.logging`.
+   * The method then:
+   *   Injects `options.logging` if called from within the tests.
+   *   Throws if called from within Sequelize and not passed correct `options.logging`
+   *
+   * @param {Object} obj Object which is parent of this method
+   * @param {String} name Name of method on object to shim
+   * @param {Integer} index Index of argument which is `options` (1-based)
+   * @param {Function} conform Function to conform function arguments
+   */
+  function shim(obj, name, index, conform) {
+    index--;
+
+    shimMethod(obj, name, function(original) {
+      return function() {
+        var args = Sequelize.Utils.sliceArgs(arguments),
+          fromTests = calledFromTests();
+
+        if (conform) args = conform.apply(this, arguments);
+
+        if (fromTests) {
+          args[index] = addLogger(args[index]);
+        } else {
+          testLogger(args[index]);
+        }
+
+        var result;
+        // NB next line written as a single statement to avoid bug with uncaught rejection
+        return (result = original.apply(this, args)) instanceof Sequelize.Promise ?
+          result.finally(finish) :
+          finish();
+
+        function finish() {
+          if (fromTests) removeLogger(args[index]);
+          return result;
+        }
+      };
+    });
+  }
+
+  /*
+   * Shims a method with given wrapper function
+   *
+   * @param {Object} obj Object which is parent of this method
+   * @param {String} name Name of method on object to shim
+   * @param {Function} wrapper Wrapper function
+   */
+  function shimMethod(obj, name, wrapper) {
+    var original = obj[name];
+    if (original.__testShim) return;
+
+    if (original.__testShimmedTo) {
+      obj[name] = original.__testShimmedTo;
+    } else {
+      obj[name] = wrapper(original);
+      obj[name].__testShim = original;
+      original.__testShimmedTo = obj[name];
+    }
+  }
```

Comment:
@mickhansen Yes, I used `shimmer` originally, but I needed to add some customizations (`__testShim` and `__testShimmedTo`), so in the end it was easier to remove `shimmer`.